src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -152,6 +153,15 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(TargetPlatformIdentifier)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+
+    <ItemGroup Condition="'$(GenerateDisableRuntimeMarshallingAttribute)' == 'true'
+                         and '$(DisableRuntimeMarshalling)' == 'true'
+                         and '$(TargetPlatformIdentifier)' != ''
```

Comment:
Is the `TargetPlatformIdentifier` clause supposed to be in here?  It seems like that means the attribute won't be written on `net7.0` for example, the TargetFramework would need to be something like `net7.0-windows` for it to be written.

There are some test failures in CI, maybe this is why they are failing.